crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -287,6 +289,40 @@ impl Timeline {
         self.inner.update_event_send_state(&txn_id, send_state).await;
     }
 
+    /// Sends an attachment to the room. It does not currently support local
+    /// echoes
+    ///
+    /// If the encryption feature is enabled, this method will transparently
+    /// encrypt the room message if the room is encrypted.
+    ///
+    /// # Arguments
+    ///
+    /// * `url` - The url for the file to be sent
+    ///
+    /// * `mime_type` - The attachment's mime type
+    ///
+    /// * `config` - An attachment configuration object containing details about
+    ///   the attachment
+    /// like a thumbnail, its size, duration etc.
+    pub async fn send_attachment(
+        &self,
+        url: String,
+        mime_type: Mime,
+        config: AttachmentConfig,
+    ) -> Result<(), Error> {
+        // If this room isn't actually in joined state, we'll get a server error.
+        // Not ideal, but works for now.
+        let room = Joined { inner: self.room().clone() };
+
+        let body =
+            Path::new(&url).file_name().ok_or(Error::InvalidAttachmentFileName)?.to_str().unwrap();
+        let data = fs::read(&url).map_err(|_| Error::InvalidAttachmentData)?;
+
+        _ = room.send_attachment(body, &mime_type, data, config).await;
```

Comment:
Why are errors from this ignored?

Reply:
Ups, my absolute bad. The initial version wasn't returning any errors and I completely missed it later. Fixed now 👍